src/rp2_common/pico_cyw43_driver/btstack_hci_transport_cyw43.c

```diff
@@ -6,21 +6,34 @@
 
 #include "pico.h"
 #include "cyw43.h"
-#include "hci_transport.h"
+#include "btstack_config.h"
 #include "hci.h"
+#include "hci_transport.h"
 #include "pico/btstack_hci_transport_cyw43.h"
 #include "pico/btstack_chipset_cyw43.h"
 
+// cyw43_bluetooth_hci_write and cyw43_bluetooth_hci_read require a custom 4-byte packet header in front of the actual HCI packet
+// the HCI packet type is stored in the fourth byte of the packet header
+#define CYW43_PACKET_HEADER_SIZE 4
+
 // assert outgoing pre-buffer for cyw43 header is available
-#if !defined(HCI_OUTGOING_PRE_BUFFER_SIZE) || (HCI_OUTGOING_PRE_BUFFER_SIZE < 4)
-#error HCI_OUTGOING_PRE_BUFFER_SIZE not defined or smaller than 4. Please update btstack_config.h
+#if !defined(HCI_OUTGOING_PRE_BUFFER_SIZE) || (HCI_OUTGOING_PRE_BUFFER_SIZE < CYW43_PACKET_HEADER_SIZE)
+#error HCI_OUTGOING_PRE_BUFFER_SIZE not defined or smaller than 4 (CYW43_PACKET_HEADER_SIZE) bytes. Please update btstack_config.h
 #endif
 
 // assert outgoing packet fragments are word aligned
 #if !defined(HCI_ACL_CHUNK_SIZE_ALIGNMENT) || ((HCI_ACL_CHUNK_SIZE_ALIGNMENT & 3) != 0)
 #error HCI_ACL_CHUNK_SIZE_ALIGNMENT not defined or not a multiply of 4. Please update btstack_config.h
 #endif
 
+// assert incoming pre-buffer for cyw43 header is available
+#if !defined(HCI_INCOMING_PRE_BUFFER_SIZE) || (HCI_INCOMING_PRE_BUFFER_SIZE < CYW43_PACKET_HEADER_SIZE)
+#error HCI_INCOMING_PRE_BUFFER_SIZE not defined or smaller than 4 (CYW43_PACKET_HEADER_SIZE) bytes. Please update btstack_config.h
+#endif
+
+// assert buffer for cyw43_bluetooth_hci_read starts word aligned (word align pre buffer)
+#define HCI_INCOMING_PRE_BUFFER_SIZE_ALIGNED ((HCI_INCOMING_PRE_BUFFER_SIZE + 3) & ~3)
```

Comment:
There's no "assert" here, perhaps "ensure" would be a more appropriate description?